### DISCUSSION TITLE
Clarification on String Set Type Requirement for fields in GraphQL/DynamoDB Tables

### DISCUSSION THREAD

**Discussion by johnnyhchen** [gp-seattle/rms-app#96] [Backend Questions]
What do we need to be a String Set type? Only in DDB storage or do we also need to be able to return a string set type (From my research so far, I think we can store it as a string set but the returned type would have to match the schema and be an array/list of strings)

**Comment by yauj** [comment]
We need to return as a string set type, since that is what the schema is expecting. Like for the tag table for example, the full form of the tags table is:
```
{
  "key": string,
  "val": {
    type: 'String',
    values: Array<string>;
  }
}
```
Dk if that answered your question.

**Reply by johnnyhchen** [reply]
I am a bit confused on how this is different from what the schema currently has. In the current schema.graphql file, the schema for the main and history table has fields which are of type arrays of strings. The corresponding fields in the Schema.ts file has type StringSet (which, when I looked at the definition, was an interface with an array of strings under the hood).

Would the current set up not return an array of strings for such a query if we connect each table to its proper DynamoDB table (instead of the autogenerated ones)?

**Reply by yauj** [reply]
Here's the thing that I worry about:
GraphQL is going to give us this back:
```
{
  "key": string,
  "val": Array<string>
}
```
while what we need is
```
{
  "key": string,
  "val": {
    type: 'String',
    values: Array<string>;
  }
}
```

I might be wrong though, so you should look up how you read a GraphQL string array to become a `StringSet`.
